04 present results

bill-amadio opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

The changes in this pull request are two new files: a notebook presenting results of the analysis and a README documenting the analysis.

#### What scientific question is your analysis addressing?

The pipeline is a response to issue #84 which was raised to check whether, in some histologies, silencing might be breaking down, potentially resulting in changes in X inactivation. Based on the accuracy achieved here, this is probably not happening, and this classifier can be helpful to predict values for data sets without annotated sex information.


#### What was your approach?

Elastic Net Logistic Regression

#### What GitHub issue does your pull request address?

84

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

Accuracy and completeness.  Conformity with standard practice.  See below,

#### Which areas should receive a particularly close look?

I had two tedious syntactical issues.  First, I could not call rmarkdown with parameters from the shell script.  I used arguments instead, which work, but I am not sure this fits standard practice.

Second, I needed to pass a shell script array to rmarkdown as an argument, but could not get that to work.  I used a shell script loop to concatenate the array values into a single string, passed that, and then unpacked the string in the notebook code.  Again, it works; not sure it is standard practice.


#### Is there anything that you want to discuss further?

I did know how to upload the results .html document to the Open PBTA collection, and so the .html document name is not linked to anything in the README file.

The line to install e1071 is still in Dockerfile.


#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

### Results

The sex prediction model achieved textbook accuracy which allays earlier concerns that silencing might be breaking down in some histologies.  This classifier can be helpful to predict values for data sets without annotated sex information.

#### What types of results are included (e.g., table, figure)?

Confusion matrix, ROC analysis tables.  Strength of calls plot along with accuracy vs. # training transcripts and # non-zero feature vs. # training transcripts.

#### What is your summary of the results?

See Results section.

#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [ X] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [X ] This analysis has been added to continuous integration.
- [X ] This analysis is recorded in the table in `analyses/README.md`.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
-->

jaclyn-taroni commented:
Hi @bill-amadio, I am going to take a look at this today! I noticed that steps 1-3 of your pipeline were commented out in the shell script. I assume that this is left over from some local development you were doing and I also suspect that this might the reason for the CI failure but I’m not entirely sure.

jaclyn-taroni commented:
I've uncommented those steps in https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404/commits/93076e1f416474586487eb78d0df69ea4ef595f8 to see what happens with continuous integration.

bill-amadio commented:
Ugh!  I apologize for making more work for you.  You are correct.  I
commented out 01-03 to see some final edits I made to the 04 notebook.

Regarding the discussion of the caret package fails, I put the 03 saves and
writes in tryCatch statements, and bypassed any writes for objects that
threw errors.  You'll find two twoClassSummary files were trapped for the
two smallest numbers of training transcripts.  All confusion matrix objects
were saved.

On Wed, Jan 8, 2020 at 7:09 AM Jaclyn Taroni <notifications@github.com>
wrote:

> Hi @bill-amadio <https://github.com/bill-amadio>, I am going to take a
> look at this today! I noticed that steps 1-3 of your pipeline were
> commented out in the shell script. I assume that this is left over from
> some local development you were doing and I also suspect that this might
> the reason for the CI failure but I’m not entirely sure.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404?email_source=notifications&email_token=ACYCOY7JZXLK5H7MIRAB5JLQ4W7BJA5CNFSM4KC5YIYKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIMFTBI#issuecomment-572021125>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACYCOY54MQBI4IHHPT5JCMLQ4W7BJANCNFSM4KC5YIYA>
> .
>

jaclyn-taroni commented:
No worries! 

One thing I am thinking about today — In some instances, we have created a separate shell script just for running things in CI that way we can make sure each individual step works without muddying up the logic for the analysis itself: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/transcriptomic-dimension-reduction/ci-dimension-reduction-plots.sh

The most recent failure I think is different than the twoClassSummary problem but I have not fully dug into it.

jaclyn-taroni commented:
Copying that error here for more context: 

```
Start script 02-train_elasticnet at 2020-01-08 13:39:59 
Loading required package: Matrix
Loading required package: foreach
Loaded glmnet 2.0-18


Attaching package: ‘glmnetUtils’

The following objects are masked from ‘package:glmnet’:

    cv.glmnet, glmnet

[1] "training set found!"
[1] "targets set found!"
[1] "target column found!"
[1] "train set sample ID sequence matches targets set sample ID sequence.  Proceeding to filtering training set columns."
[1] "MAD filtering is complete.  train_set dimension =  13 X 3127 . Proceeding to model build."
[1] "Model build begun at 2020-01-08 13:40:12"
Error in lognet(x, is.sparse, ix, jx, y, weights, offset, alpha, nobs,  : 
  one multinomial or binomial class has 1 or 0 observations; not allowed
Calls: cva.glmnet ... cva.glmnet.default -> lapply -> FUN -> <Anonymous> -> glmnet -> lognet
In addition: Warning messages:
1: In lognet(x, is.sparse, ix, jx, y, weights, offset, alpha, nobs,  :
  one multinomial or binomial class has fewer than 8  observations; dangerous ground
2: In lognet(x, is.sparse, ix, jx, y, weights, offset, alpha, nobs,  :
  one multinomial or binomial class has fewer than 8  observations; dangerous ground
3: In lognet(x, is.sparse, ix, jx, y, weights, offset, alpha, nobs,  :
  one multinomial or binomial class has fewer than 8  observations; dangerous ground
4: In lognet(x, is.sparse, ix, jx, y, weights, offset, alpha, nobs,  :
  one multinomial or binomial class has fewer than 8  observations; dangerous ground
5: In lognet(x, is.sparse, ix, jx, y, weights, offset, alpha, nobs,  :
  one multinomial or binomial class has fewer than 8  observations; dangerous ground
6: In lognet(x, is.sparse, ix, jx, y, weights, offset, alpha, nobs,  :
  one multinomial or binomial class has fewer than 8  observations; dangerous ground
```

jaclyn-taroni commented:
I'm going to try adding some logic to `run-sex-prediction-from-RNASeq.sh` here rather than go the separate shell script route.

jaclyn-taroni commented:
Okay @bill-amadio - what I did in https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404/commits/61cc9df9b50295e001a6c42b4e5e674cff9756bc is use an environmental variable just for continuous integration. Specifically, it's set up such that when this shell script is run in CI, it will only test one value for MAD filtering. That is accomplished with this logic:

```sh
if ["$PERCENT" -gt "0"]; then
  TRANSCRIPT_TAIL_PERCENT_ARRAY=(0.25 0.125 0.0625 0.0313 0.0156 0.0078 0.0039 0.0020 0.0010 0.0005)
else
  TRANSCRIPT_TAIL_PERCENT_ARRAY=(0.25)
fi
```

Where `$PERCENT` will be `1` (and therefore greater than `0`) by default: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404/files#diff-b2284d881a9b7b77449ca64573a70d9eR17

And we set `$PERCENT` to `0` in CI: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404/files#diff-1d37e48f9ceff6d8030570cd36286a61R47

More info here: https://github.com/AlexsLemonade/OpenPBTA-analysis#passing-variables-only-in-ci

I've made these changes so I can focus my review on correctness and completeness for the full dataset.

jaclyn-taroni commented on analyses/sex-prediction-from-RNASeq/01-clean_split_data.R:
```diff
@@ -101,6 +101,12 @@ option_list <- list(
+    c("-e", "--target_columns"),
+    type = "character",
+    default = "reported_gender",
+    help = "Names of columns to use as targets in testing",
```
```suggestion
    help = "Names of columns to use as targets in testing separated with a comma",
```
because it should be something like `reported_gender,germline_sex_estimate` correct?

bill-amadio replied:
Yes.  The shell script  concatenates all columns specified and passes a single string with values separated by commas for this argument.

Will do in local repository and push.

jaclyn-taroni commented on analyses/sex-prediction-from-RNASeq/02-train_elasticnet.R:
```diff
@@ -145,7 +145,7 @@ print(paste("Model build begun at", Sys.time(), sep=" "))
 
 sex.cva <- cva.glmnet(train_set, targets_set[, opt$train_target_column], standardize=TRUE,
-                      alpha = seq(0, 1, len = 11)^3, family="binomial")
+                      alpha = seq(0, 1, len = 11)^3, foldid=sample(1:10,size=nrow(train_set),replace=TRUE), family="binomial")
```
What is the significance of this change?

bill-amadio replied:
Once I saw the variance in the number of non-zero features vs. # training transcripts, I thought it might be useful to have this in case someone wanted to do some fine-tuning of the elastic net parameters.

jaclyn-taroni commented on analyses/sex-prediction-from-RNASeq/run-sex-prediction-from-RNASeq.sh:
```diff
@@ -62,59 +90,76 @@ Rscript --vanilla 01-clean_split_data.R \
-MODEL_OBJECT_FILE_NAME=${FILENAME_LEAD}_${SEED}_${TRANSCRIPT_TAIL_PERCENT}_model_object.RDS
-MODEL_TRANSCRIPTS_FILE_NAME=${FILENAME_LEAD}_${SEED}_${TRANSCRIPT_TAIL_PERCENT}_model_transcripts.RDS
-MODEL_COEFS_FILE_NAME=${FILENAME_LEAD}_${SEED}_${TRANSCRIPT_TAIL_PERCENT}_model_coefs.tsv
+MODEL_OBJECT_FILE_NAME=${FILENAME_LEAD}_${SEED}_${TRANSCRIPT_TAIL_PERCENT_ARRAY[i]}_model_object.RDS
```
Can you indent everything in this `for` loop to make it clear that it is in a loop
```suggestion
  MODEL_OBJECT_FILE_NAME=${FILENAME_LEAD}_${SEED}_${TRANSCRIPT_TAIL_PERCENT_ARRAY[i]}_model_object.RDS
```

bill-amadio replied:
Will do all lines on my local repositiory and push.

jaclyn-taroni commented:
Hi @bill-amadio - thanks for sending this PR! To respond to the areas that need a closer look

> I had two tedious syntactical issues. First, I could not call rmarkdown with parameters from the shell script. I used arguments instead, which work, but I am not sure this fits standard practice.

You can [use parameters with Rmd files](https://rmarkdown.rstudio.com/lesson-6.html). I've added line comments in the pending review that I will send shortly to illustrate how you can use them here.

> Second, I needed to pass a shell script array to rmarkdown as an argument, but could not get that to work. I used a shell script loop to concatenate the array values into a single string, passed that, and then unpacked the string in the notebook code. Again, it works; not sure it is standard practice.

I believe you are talking about this line https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404/files#diff-b2284d881a9b7b77449ca64573a70d9eR161 and then you use `strsplit` which seems reasonable to me.

bill-amadio commented on analyses/sex-prediction-from-RNASeq/README.md:
```diff
@@ -0,0 +1,226 @@
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  
+
+- [How to run sex prediction from RNASeq](#how-to-run-sex-prediction-from-RNASeq)
```
I've pulled down your changes, and I'm going through your suggestions.    The links are working as is, but I can certainly change them if necessary.

jaclyn-taroni replied:
The rich diff for this markdown file looks like this for me:

<img width="959" alt="Screen Shot 2020-01-09 at 1 42 39 PM" src="https://user-images.githubusercontent.com/19534205/72095182-feaef580-32e5-11ea-89c7-3279f12cd519.png">

which is what makes me think the uppercase letters are causing issues.

bill-amadio replied:
Links are working.  I can change if necessary.

bill-amadio commented on analyses/sex-prediction-from-RNASeq/README.md:
```diff
@@ -0,0 +1,226 @@
+
+**Overall notes about these scripts:**
+- The scripts are sequential as noted by their number.
+- All file path-related options assume the file path given is relative to `OpenPBTA-analysis/analyses/sex-prediction-from-RNASeq`.
```
I meant to communicate that if we use --output_directory processed_data as an argument for script 01, the output files of the script will go to OpenPBTA-analysis/analyses/sex-prediction-from-RNASeq/processed_data.

Does this work better: All file path-related arguments assume the file path specified is relative to `OpenPBTA-analysis/analyses/sex-prediction-from-RNASeq`?

jaclyn-taroni replied:
Yes, that wording sounds good.

bill-amadio commented on analyses/sex-prediction-from-RNASeq/run-sex-prediction-from-RNASeq.sh:
```diff
@@ -62,59 +90,76 @@ Rscript --vanilla 01-clean_split_data.R \
+
+
+
+Rscript -e "rmarkdown::render('04-present_results.Rmd', 'html_document')" --args \
```
Still getting an error:
Error in knit_params_get(input_lines, params) :
  object 'results' not found
Calls: <Anonymous> -> knit_params_get
Execution halted

The notebook with params works fine interactively.  I tried a two-line script and simple notebook with just the assignment to $RESULTS, and call 
Rscript -e "rmarkdown::render('simple.Rmd', 'html_document', params = list(results_dir = ${RESULTS}))"

Same results.  Search for the error message did not return any similar situations.  Could it be I'm running on a windows machine?

jaclyn-taroni replied:
If you try:
```
Rscript -e "rmarkdown::render('simple.Rmd', 'html_document', params = list(results_dir = 'results'))"
```
does that work? Trying to narrow things down.

jaclyn-taroni replied:
Okay @bill-amadio - I think single quotes around `${RESULTS}` should solve it, my apologies for omitting them - does the following work?
```
Rscript -e "rmarkdown::render('simple.Rmd', 'html_document', params = list(results_dir = '${RESULTS}'))"
```

bill-amadio commented on analyses/sex-prediction-from-RNASeq/README.md:
```diff
@@ -0,0 +1,226 @@
+# Sex Prediction from RNASeq
+
+The 01-clean_split_data.R, 02-train_elasticnet.R, 03-evaluate_model.R, 04-present_results.Rmd pipeline trains and evaluates an elasticnet logistic regression model to predict sex from RNASeq data.  The training features are gene expression transcripts, and the training labels are reported_gender values for each sample.
```
Will do in my local repository and push changes.

bill-amadio commented:
I will make the changes in my local repository and push.

On Thu, Jan 9, 2020 at 1:44 PM Jaclyn Taroni <notifications@github.com>
wrote:

> *@jaclyn-taroni* commented on this pull request.
> ------------------------------
>
> In analyses/sex-prediction-from-RNASeq/README.md
> <https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404#discussion_r364900403>
> :
>
> > @@ -0,0 +1,226 @@
> +# Sex Prediction from RNASeq
> +
> +The 01-clean_split_data.R, 02-train_elasticnet.R, 03-evaluate_model.R, 04-present_results.Rmd pipeline trains and evaluates an elasticnet logistic regression model to predict sex from RNASeq data.  The training features are gene expression transcripts, and the training labels are reported_gender values for each sample.
> +
> +The pipeline is a response to issue [#84](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/84) which was raised to check whether, in some histologies, silencing might be breaking down, potentially resulting in changes in X inactivation. Based on the accuracy achieved here, this is probably not happening, and this classifier can be helpful to predict values for datasets without annotated sex information.
> +
> +<!-- START doctoc generated TOC please keep comment here to allow auto update -->
> +<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
> +**Table of Contents**
> +
> +- [How to run sex prediction from RNASeq](#how-to-run-sex-prediction-from-RNASeq)
>
> The rich diff for this markdown file looks like this for me:
>
> [image: Screen Shot 2020-01-09 at 1 42 39 PM]
> <https://user-images.githubusercontent.com/19534205/72095182-feaef580-32e5-11ea-89c7-3279f12cd519.png>
>
> which is what makes me think the uppercase letters are causing issues.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404?email_source=notifications&email_token=ACYCOYZHCS2367WQENAQS73Q45V67A5CNFSM4KC5YIYKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCRHTB7I#discussion_r364900403>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACYCOY7LETI7VVZLUDYHOWTQ45V67ANCNFSM4KC5YIYA>
> .
>

bill-amadio commented:
Yes.  That works.  And so does Rscript -e "rmarkdown::render('simple.Rmd',
'html_document', params = list(results_dir = 'results', second_line =
'results 2'))" so it is not the number of parameters.



On Thu, Jan 9, 2020 at 8:37 PM Jaclyn Taroni <notifications@github.com>
wrote:

> *@jaclyn-taroni* commented on this pull request.
> ------------------------------
>
> In analyses/sex-prediction-from-RNASeq/run-sex-prediction-from-RNASeq.sh
> <https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404#discussion_r365039398>
> :
>
> >    done
>  fi
>
> +
> +done
> +
> +Rscript -e 'library(rmarkdown)'
> +
> +
> +
> +Rscript -e "rmarkdown::render('04-present_results.Rmd', 'html_document')" --args \
>
> If you try:
>
> Rscript -e "rmarkdown::render('simple.Rmd', 'html_document', params = list(results_dir = 'results'))"
>
> does that work? Trying to narrow things down.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404?email_source=notifications&email_token=ACYCOY7L7R6GJL2CF2K7KVLQ47GM3A5CNFSM4KC5YIYKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCRI7EFI#discussion_r365039398>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACYCOY4ANVXBKJOFZBRTOYDQ47GM3ANCNFSM4KC5YIYA>
> .
>

jaclyn-taroni commented:
@bill-amadio can you push the changes to the `analyses/sex-prediction-from-RNASeq/04-present_results.Rmd` notebook (specifically the addition of `params` to the YAML header) so I can take a look locally please?

bill-amadio commented:
done.

On Fri, Jan 10, 2020 at 8:45 AM Jaclyn Taroni <notifications@github.com>
wrote:

> @bill-amadio <https://github.com/bill-amadio> can you push the changes to
> the analyses/sex-prediction-from-RNASeq/04-present_results.Rmd notebook
> (specifically the addition of params to the YAML header) so I can take a
> look locally please?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404?email_source=notifications&email_token=ACYCOY7QDQMAYWIEJ3TBBALQ5B3XTA5CNFSM4KC5YIYKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIT6TJI#issuecomment-573041061>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACYCOYZHLQ54X5ZU3W7RHLTQ5B3XTANCNFSM4KC5YIYA>
> .
>

bill-amadio commented:
Yeah!!!

It works for the simple script and for 04-present_results.  The example you
cited,
https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/7917a7fc9d4328e8fc8142caca635e5c2e73df97/analyses/gene-set-enrichment-analysis/run-gsea.sh#L43,
does not use the quotes.  Curious.

Thank you so much for tracking down the solution.  I cannot tell you how
many attempts I made both before and after the PR.

Can I push the rest of my changes  today or tomorrow?  I'll check, but I
believe I've answered all questions and addressed all change requests.  Are
you satisfied with why I added the foldids parameter to the model call?  I
don't want to create any conflicts with changes you might be making.





On Fri, Jan 10, 2020 at 11:39 AM Jaclyn Taroni <notifications@github.com>
wrote:

> *@jaclyn-taroni* commented on this pull request.
> ------------------------------
>
> In analyses/sex-prediction-from-RNASeq/run-sex-prediction-from-RNASeq.sh
> <https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404#discussion_r365324506>
> :
>
> >    done
>  fi
>
> +
> +done
> +
> +Rscript -e 'library(rmarkdown)'
> +
> +
> +
> +Rscript -e "rmarkdown::render('04-present_results.Rmd', 'html_document')" --args \
>
> Okay @bill-amadio <https://github.com/bill-amadio> - I think single
> quotes around ${RESULTS} should solve it, my apologies for omitting them
> - does the following work?
>
> Rscript -e "rmarkdown::render('simple.Rmd', 'html_document', params = list(results_dir = '${RESULTS}'))"
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404?email_source=notifications&email_token=ACYCOY43JSGUTGCSBNLRXT3Q5CQFBA5CNFSM4KC5YIYKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCRLZKLQ#discussion_r365324506>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACYCOYYEASDJ3UIHZEESMPLQ5CQFBANCNFSM4KC5YIYA>
> .
>

jaclyn-taroni commented:
Woo! Glad that worked for you.

> The example you
> cited,
> https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/7917a7fc9d4328e8fc8142caca635e5c2e73df97/analyses/gene-set-enrichment-analysis/run-gsea.sh#L43,
> does not use the quotes.  Curious.

I'm pretty sure that works because it's an integer that gets turned into a logical. 

> Can I push the rest of my changes  today or tomorrow?  I'll check, but I
believe I've answered all questions and addressed all change requests.  Are
you satisfied with why I added the foldids parameter to the model call?  I
don't want to create any conflicts with changes you might be making.

Yes, this sounds good. I did not go through the `analyses/sex-prediction-from-RNASeq/04-present_results.Rmd` notebook yet. I wanted to address some of your questions first. So we will most likely have another round of back and forth.

bill-amadio commented on analyses/sex-prediction-from-RNASeq/04-present_results.Rmd:
````diff
@@ -0,0 +1,249 @@
+
+```{r echo=FALSE}
+
+#for each value in targetColumns argument, find the index of the row corresponding to the maximum predictive accuracy
````
I thought we should have one copy of this plot because we only trained on the reported_gender values.  We can calculate values of maximum_accuracy_index for each target_column value, and use them in this plot and the subsequent plots, but if we are going to vary the model trained on reported_gender values according to test target column, shouldn't we just train on germline_sex_estimate instead when those values are available?

jaclyn-taroni replied:
The single plot sounds good. I think this is a matter of my interpretation of the comment not matching the implementation/functionality of the code.

bill-amadio replied:
changed the comment and the logic of the data supporting the plot in the latest push.  Still just one copy of Strength of Calls.

bill-amadio replied:
To check reproducibility, I looked at model features and coefficients for two separate runs of the pipeline.  They agree.

In the Strength of Calls plot, there are differences in appearance for two separate runs of the pipeline because ggplot varies horizontal placement of points in the dot plot.  For two separate runs of the plot, there is, however, a 1-to-1 correspondence between the y-values.  So, the same result is being reported in the two plots.

bill-amadio commented on analyses/sex-prediction-from-RNASeq/04-present_results.Rmd:
```diff
@@ -0,0 +1,249 @@
+  number_non_zero <- data.frame(ID=transcript_tail_percents, number_features=number_non_zero)
+
+#join # non-zero features to predictive accuracies and # training transcript values by ID  
+  final_df <- suppressWarnings(inner_join(pre_final_df, number_non_zero, by="ID"))
```
My first guess (and hope!) is that the intercept is included in the non-zero features.  Will check the files today.

bill-amadio replied:
The intercept is the problem.  Script 02 needs a quick fix.

We identify non-zero features with

non_zero_features <- which(coef(best_fit, s = best_fit$lambda.1se) != 0)

A look into coef(best_fit, s = best_fit$lambda.1se) returns

202 x 1 sparse Matrix of class "dgCMatrix"
                                             1
(Intercept)                       1.130771e-01
ENST00000009589.7_RPS20-201      -7.916031e-41
ENST00000084795.9_RPL18-201      -8.778138e-41
.
.
.

So, we need to ignore the first non-zero coefficient when identifying non-zero TRANSCRIPTS.

How did you find this?

jaclyn-taroni replied:
This point was not necessarily consistent with my expectations + looking locally at `df_list`
<img width="690" alt="Screen Shot 2020-01-15 at 11 48 50 AM" src="https://user-images.githubusercontent.com/19534205/72453672-7c16b200-378d-11ea-91d0-d86654a781d9.png">

Mostly looking at the tables locally, though.

bill-amadio replied:
fix is in latest push.

bill-amadio commented on analyses/sex-prediction-from-RNASeq/04-present_results.Rmd:
```diff
@@ -0,0 +1,249 @@
+params:
+  results_dir: "results"
+  model_dir: "models"
+  cm_set: NULL
```
I removed it.

bill-amadio commented on analyses/sex-prediction-from-RNASeq/04-present_results.Rmd:
```diff
@@ -0,0 +1,249 @@
+  files_to_process <- list.files(path = results_directory, pattern = "_confusion_matrix.RDS")
+  first_pos <- sapply(files_to_process, function(x) regexpr(params$seed, x))
+  last_pos <- sapply(files_to_process, function(x) regexpr("confusion_matrix", x))
+  transcript_tail_percents <- substr(files_to_process, 1, last_pos-2)
```
I will look at how to use tidyverse and stringr here.

jaclyn-taroni replied:
Everything seems to be _correct_. I am running the whole pipeline now so I can take another look locally. I want to get this merged so that the module has a README in the master branch and then any improvements of this nature can go into a subsequent pull request. In general, we aim to keep pull requests focused on a single file or area per [the contributing guidelines](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/CONTRIBUTING.md#size-and-composition-of-pull-requests) which helps ensure a thorough review can be completed in a timely manner. So I think this is a case where we want to get everything you've done in and then we can refine things as needed.

bill-amadio commented:
Glad to see it in the master.  I am working through the stringr tutorial at
https://cran.r-project.org/web/packages/stringr/vignettes/stringr.html.
Also, I need to get TocDoc competent.  The copy and paste site produced a
blank screen, which others have reported on SO.

On Tue, Jan 21, 2020 at 10:28 AM Jaclyn Taroni <notifications@github.com>
wrote:

> Merged #404 <https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404>
> into master.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/404?email_source=notifications&email_token=ACYCOYZMIRSQWMUPSPUIETLQ64IARA5CNFSM4KC5YIYKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOWDJXAAA#event-2966646784>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACYCOYZHT2TSGNSRPUQEL7DQ64IARANCNFSM4KC5YIYA>
> .
>
